src/Microsoft.AspNet.Mvc.Core/ModelBinding/BodyModelBinder.cs

```diff
@@ -84,9 +84,11 @@ public Task<ModelBindingResult> BindModelAsync(ModelBindingContext bindingContex
 
             if (formatter == null)
             {
-                var unsupportedContentType = Resources.FormatUnsupportedContentType(
+                var message = Resources.FormatUnsupportedContentType(
                     bindingContext.OperationBindingContext.HttpContext.Request.ContentType);
-                bindingContext.ModelState.AddModelError(modelBindingKey, unsupportedContentType);
+
+                var exception = new UnsupportedContentTypeException(message);
+                bindingContext.ModelState.AddModelError(modelBindingKey, exception, bindingContext.ModelMetadata);
```

Comment:
Sounds good. One clarification:
> it's not an error users will make

That's not relevant because it is an error users could _see_ in their browser whenever `UnsupportedContentTypeFilter` is disabled. `ModelError.ErrorMessage` is for display to users regardless of who or what caused the error.